### ISSUE TITLE
No wayland support on Linux (missing platform plugin)

### ISSUE THREAD

**Issue by IntelligentesTierMaulApollo13** [gitahead/gitahead#275]
On newer Ubuntu (at least 18.04+) wayland is the default gnome session type. If one tries to execute GitAhead as root on such a platform Qt complains about a missing platform plugin ("wayland"). Can you please add that support?

In my tests (with own applications) adding libwayland-generic.so was enough. Perhaps it is enough for GitAhead as well?

**Comment by silverscania** [comment]
I just installed from [this link](https://github.com/gitahead/gitahead/releases/download/v2.5.10/GitAhead-2.5.10.sh) on Ubuntu 19 and it worked fine.

**Comment by IntelligentesTierMaulApollo13** [comment]
Did you execute it as root (e.g. via sudo)? Using GitAhead without root (i.e. running under my regular user) works flawlessly, but as far as I know that is due to the fact that XWayland can be used then (and thus the xcb platform plugin is enough). Running as root, XWayland is unavailable, however, so Qt has to use the wayland plugin.

**Comment by silverscania** [comment]
No, I don't normally have much luck running GUI applications as root, even X11 ones. So I wouldn't expect it to work and wouldn't recommend it. Why are you trying to run it that way?

Have you tried running it with sudo and preserving your environment?
> sudo -EH ./GitAhead
